### PR TITLE
Remove remaining deprecated APIs

### PR DESCRIPTION
**Pull request by bluwy**
## Changes

The changesets should explain the remaining various removals:

> - The Astro preview server now returns a 404 status instead of a 301 redirect when requesting assets from the public directory without a base.
> - Removes special handling when referencing the `astro/client-image` type. You should use the `astro/client` type instead.
> - Removes deprecated builtin `rss` support. You should use `@astrojs/rss` instead.
> - Removes deprecated `Astro.request.params` support. You should use `Astro.params` instead.
> - Remove `@astrojs/markdoc` compat support for Astro 3
> - Removes the deprecated `@astrojs/vercel/edge` export.

## Testing

Existing tests should pass

## Docs

Added changesets and deprecated one error code.

### CODE REVIEW

File: packages/astro/src/core/build/plugins/index.ts [hunk 1]
```diff
@@ -18,7 +17,6 @@ import { pluginSSR, pluginSSRSplit } from './plugin-ssr.js';
 
 export function registerAllPlugins({ internals, options, register }: AstroBuildPluginContainer) {
 	register(pluginComponentEntry(internals));
-	register(pluginAliasResolve(internals));
```
Comment (bluwy) [inline packages/astro/src/core/build/plugins/index.ts:21]: This is a silent change that's not documented in the changesets as it's fixed from a Vite 5 upgrade. Users shouldn't notice importing components with aliases not working.

File: packages/astro/src/core/errors/errors-data.ts [hunk 2]
```diff
@@ -292,6 +292,7 @@ export const InvalidGetStaticPathsReturn = {
 
 /**
  * @docs
+ * @deprecated Deprecated since Astro 4.0. The RSS helper no longer exists with an error fallback.
```
Comment (Princesseuh) [inline packages/astro/src/core/errors/errors-data.ts:295]: 🤩

**Review by natemoo-re — APPROVED** [review]
These changes all look good to me!

**Review by ematipico — APPROVED** [review]
Done! 

Mic drop

File: .changeset/green-parrots-brake.md [hunk 3]
```diff
@@ -0,0 +1,10 @@
+---
+'astro': major
+---
+
+Removes additional deprecated APIs:
+
+- The Astro preview server now returns a 404 status instead of a 301 redirect when requesting assets from the public directory without a base.
+- Removes special handling when referencing the `astro/client-image` type. You should use the `astro/client` type instead.
+- Removes deprecated builtin `rss` support. You should use `@astrojs/rss` instead.
```
Comment (sarah11918) [inline .changeset/green-parrots-brake.md:9]:
```suggestion
- Removes deprecated built-in `rss` support. You should use `@astrojs/rss` instead.
```

**Review by sarah11918 — APPROVED** [review]
Looks fabulous, @bluwy !  🥳  just one typo!

File: .changeset/green-parrots-brake.md [hunk 4]
```diff
@@ -0,0 +1,10 @@
+---
+'astro': major
+---
+
+Removes additional deprecated APIs:
+
+- The Astro preview server now returns a 404 status instead of a 301 redirect when requesting assets from the public directory without a base.
+- Removes special handling when referencing the `astro/client-image` type. You should use the `astro/client` type instead.
+- Removes deprecated built-in `rss` support. You should use `@astrojs/rss` instead.
```
Comment (bluwy) [inline .changeset/green-parrots-brake.md:9]:
```suggestion
- Removes deprecated built-in `rss` support in `getStaticPaths`. You should use `@astrojs/rss` instead.
```

**Comment by bluwy** [comment]
Thanks everyone for the quick reviews 🙌

**Pull request merged**